runners/google-cloud-dataflow-java/worker/src/main/java/org/apache/beam/runners/dataflow/worker/windmill/client/grpc/GrpcCommitWorkStream.java

```diff
@@ -283,21 +312,105 @@ private void issueMultiChunkRequest(final long id, PendingRequest pendingRequest
     }
   }
 
-  private static class PendingRequest {
+  /**
+   * Returns true if the request should be failed due to stream shutdown, else tracks the request to
+   * be sent and returns false.
+   */
+  private boolean isPrepareForSendFailed(long id, PendingRequest request) {
```

Comment:
nit:
```suggestion
  private boolean prepareForSend(long id, PendingRequest request) {
```

and return true if prepare succeeded.

Reply:
done